### PR TITLE
Add travis/github retry logic

### PR DESCRIPTION
**Pull request by cognifloyd**
##### SUMMARY
Implement retries from #178 

Improves things, but we're still subject to a hard-coded timeout of 20 seconds in ansible-galaxy cli. Talking with the galaxy folks, it's apparently a widespread issue right now.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
CI

### CODE REVIEW

**Comment by rhysmeister** [comment]
I'd remove the retries from the pip/collection installs. These are annoying but should be fixed at the galaxy end IMHO. I've always assumed that the delete the archive/resource prematurely during the build process, meaning it's unavailable for a short time. Often this seems to be many minutes so I'm not sure a few retries will fix these.

**Comment by cognifloyd** [comment]
Where were you thinking of putting travis_retry? pip and ansible-galaxy were the only network sensitive commands I saw.

**Comment by rhysmeister** [comment]
At least some of these timouts are due to the dependancies in community.general...

```
dependencies:
  ansible.netcommon: '>=1.0.0'
  ansible.posix: '>=1.0.0'
  community.kubernetes: '>=1.0.0'
  google.cloud: '>=1.0.0'
```

https://github.com/ansible-collections/community.general/blob/main/galaxy.yml

Probably most of these shouldn't be here. I think there was a ticket about removing some of these, for example I think google.cloud was only included because of a dep on a single function. I might chase this up.

**Comment by gundalow** [comment]
@cognifloyd @rhysmeister Hi
1. The Galaxy team at Red Hat have identified some performance issues which were resulting in galaxy.ansible.com to be unresponsive, partially on the hour. Some of these issues have been resolved, so I'd hope we see fewer timeouts
2. We are moving more content out of `community.general` so there will be fewer dependencies, see https://github.com/ansible/community/issues/539

File: .github/workflows/ansible-test.yml [hunk 1]
```diff
@@ -104,13 +104,13 @@ jobs:
...
-        run: ansible-galaxy collection install community.general -p ../../
+        run: 'for i in {1..5}; do [ "${i}" -ne 1 ] && sleep 5; ansible-galaxy collection install community.general -p ../../ && break; done'
 
       - name: Install community.crypto
-        run: ansible-galaxy collection install community.crypto -p ../../
+        run: 'for i in {1..5}; do [ "${i}" -ne 1 ] && sleep 5; ansible-galaxy collection install community.crypto -p ../../ && break; done'
 
       - name: Install ansible-base (${{ matrix.ansible_version }})
-        run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible_version }}.tar.gz --disable-pip-version-check
+        run: 'for i in {1..5}; do [ "${i}" -ne 1 ] && sleep 5; pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible_version }}.tar.gz --disable-pip-version-check && break; done'
```
Comment (gundalow) [inline .github/workflows/ansible-test.yml:113]: `ansible-base` should be installed *before* the collections, that way if we improve `ansible-galaxy collection install` you will get the benefit.

File: .github/workflows/ansible-test.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
           python-version: 3.6
 
       - name: Install ansible-base (devel)
-        run: pip install https://github.com/ansible/ansible/archive/devel.tar.gz --disable-pip-version-check
+        run: 'for i in {1..5}; do [ "${i}" -ne 1 ] && sleep 5; pip install https://github.com/ansible/ansible/archive/devel.tar.gz --disable-pip-version-check && break; done'
```
Comment (gundalow) [inline .github/workflows/ansible-test.yml:27]: Have you seen failures installing from `https://github.com/ansible/ansible/archive/devel.tar.gz`?

**Comment by rhysmeister** [comment]
@cognifloyd Timeouts are now pretty much sorted. Can we close this without merging or is there anything that needs to stay?

**Comment by rhysmeister** [comment]
Not relevant anymore. Closing.

**Pull request closed without merging**